Intermittent NullReferenceException in CSharpDiagnosticAnalyzerApiUsageAnalyzer

_From @yaakov-h on August 7, 2018 3:58_

**Version Used**: 
Microsoft.CodeAnalysis.Analyzers v2.6.1

**Steps to Reproduce**:

1. Clone https://github.com/WiseTechGlobal/WTG.Analyzers at commit 223095d2
2. Build from source

**Expected Behavior**:

No null reference exceptions in MS analyzers

**Actual Behavior**:

Sometimes null reference exception in MS analyzers

> CSC : error AD0001: Analyzer 'Microsoft.CodeAnalysis.CSharp.Analyzers.MetaAnalyzers.CSharpDiagnosticAnalyzerApiUsageAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [C:\projects\wtg-analyzers\WTG.Analyzers.Test\WTG.Analyzers.Test.csproj]

_Copied from original issue: dotnet/roslyn#29119_

@yaakov-h I am unable to build your solution:

```
1>------ Build started: Project: WTG.Analyzers.Utils, Configuration: Debug Any CPU ------
1>c:\WTG.Analyzers\WTG.Analyzers.Utils\WTG.Analyzers.Utils.csproj(11,3): error MSB4019: The imported project "c:\WTG.Analyzers\.paket\Paket.Restore.targets" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
1>Done building project "WTG.Analyzers.Utils.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

--- a/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerAPIUsageAnalyzer.cs
+++ b/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerAPIUsageAnalyzer.cs
@@ -235,7 +235,8 @@ namespace Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers
 
         private static void AddUsedNamedTypeCore(ITypeSymbol typeOpt, ImmutableHashSet<INamedTypeSymbol>.Builder builder, ref bool hasAccessToTypeFromWorkspaceAssemblies)
         {
-            if (typeOpt is INamedTypeSymbol usedType)
+            if (typeOpt is INamedTypeSymbol usedType &&
+                usedType.TypeKind != TypeKind.Error)
             {
                 builder.Add(usedType);
 

--- a/src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DiagnosticAnalyzerApiUsageAnalyzerTests.cs
+++ b/src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/DiagnosticAnalyzerApiUsageAnalyzerTests.cs
@@ -33,6 +33,7 @@ class MyAnalyzer : DiagnosticAnalyzer
 
     public override void Initialize(AnalysisContext context)
     {
+        var x = typeof(I<>);
     }
 }
 

